packages/foam-vscode/src/services/variable-resolver.ts

```diff
@@ -120,14 +122,23 @@ export class Resolver implements VariableResolver {
     return (variable.children[0] ?? name).toString();
   }
 
-  resolve(variable: Variable): Promise<string | undefined> {
+  async resolve(variable: Variable): Promise<string | undefined> {
     const name = variable.name;
     if (this.givenValues.has(name)) {
       this.promises.set(name, Promise.resolve(this.givenValues.get(name)));
     } else if (!this.promises.has(name)) {
       switch (name) {
         case 'FOAM_TITLE':
-          this.promises.set(name, resolveFoamTitle());
+          this.promises.set(
+            name,
+            Promise.resolve(await this.resolveFoamTitle())
+          );
+          break;
+        case 'FOAM_SLUG':
+          this.promises.set(
+            name,
+            Promise.resolve(toSlug(await this.resolveFoamTitle()))
+          );
```

Comment:
```suggestion
          this.promises.set(
            name,
            Promise.resolve(toSlug(await this.resolve('FOAM_TITLE')))
          );
```

I believe that if we use this way or getting the title we won't need to store it, because of the check at line 129, right before the switch statement.

Reply:
You were right so I've pushed those updates to the branch 🙌🏼